[BUG] compile error after dependency update to version 1.70.1

**Version of dd-trace-go**
1.70.1

**Describe what happened:**
After  dependency upgrades this compile error is raised from the package:
gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer
# gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer
../../../../../go/pkg/mod/gopkg.in/!data!dog/dd-trace-go.v1@v1.70.1/ddtrace/tracer/tracer.go:339:34: cannot use t.statsd (variable of type "gopkg.in/DataDog/dd-trace-go.v1/internal".StatsdClient) as runtimemetrics.partialStatsdClientInterface value in argument to runtimemetrics.Start: "gopkg.in/DataDog/dd-trace-go.v1/internal".StatsdClient does not implement runtimemetrics.partialStatsdClientInterface (missing method DistributionSamples)


**Describe what you expected:**
No compiler error. It seems to be dependency-related, but the error is from the library itself. I see no combination of dependency updates that causes this.

**Steps to reproduce the issue:**
Dependency update with `go get -u ./...` and `go mod tidy`. After this, the error above occurred in the builds.

**Additional environment details (Version of Go, Operating System, etc.):**
Go 1.23.4, across operation systems (this error also occurs on CI/Linux and on macOS)

--- a/go.mod
+++ b/go.mod
@@ -12,7 +12,7 @@ require (
 	github.com/DataDog/datadog-agent/pkg/trace v0.58.0
 	github.com/DataDog/datadog-go/v5 v5.5.0
 	github.com/DataDog/go-libddwaf/v3 v3.5.1
-	github.com/DataDog/go-runtime-metrics-internal v0.0.0-20241106155157-194426bbbd59
+	github.com/DataDog/go-runtime-metrics-internal v0.0.3
 	github.com/DataDog/gostackparse v0.7.0
 	github.com/DataDog/sketches-go v1.4.5
 	github.com/IBM/sarama v1.40.0

--- a/internal/apps/go.mod
+++ b/internal/apps/go.mod
@@ -14,7 +14,7 @@ require (
 	github.com/DataDog/datadog-agent/pkg/util/log v0.58.0 // indirect
 	github.com/DataDog/datadog-agent/pkg/util/scrubber v0.58.0 // indirect
 	github.com/DataDog/go-libddwaf/v3 v3.5.1 // indirect
-	github.com/DataDog/go-runtime-metrics-internal v0.0.0-20241106155157-194426bbbd59 // indirect
+	github.com/DataDog/go-runtime-metrics-internal v0.0.3 // indirect
 	github.com/DataDog/go-sqllexer v0.0.14 // indirect
 	github.com/DataDog/opentelemetry-mapping-go/pkg/otlp/attributes v0.20.0 // indirect
 	github.com/cihub/seelog v0.0.0-20170130134532-f561c5e57575 // indirect

--- a/internal/exectracetest/go.mod
+++ b/internal/exectracetest/go.mod
@@ -19,7 +19,7 @@ require (
 	github.com/DataDog/datadog-agent/pkg/util/scrubber v0.58.0 // indirect
 	github.com/DataDog/datadog-go/v5 v5.5.0 // indirect
 	github.com/DataDog/go-libddwaf/v3 v3.5.1 // indirect
-	github.com/DataDog/go-runtime-metrics-internal v0.0.0-20241106155157-194426bbbd59 // indirect
+	github.com/DataDog/go-runtime-metrics-internal v0.0.3 // indirect
 	github.com/DataDog/go-sqllexer v0.0.14 // indirect
 	github.com/DataDog/go-tuf v1.1.0-0.5.2 // indirect
 	github.com/DataDog/opentelemetry-mapping-go/pkg/otlp/attributes v0.20.0 // indirect

